inginious/frontend/pages/course_admin/submission_viewer.py

```diff
@@ -0,0 +1,228 @@
+        self._valid_formats_name = [_("taskid/username"), _("taskid/aggregation"), _("username/taskid"), _("aggregation/taskid")]
+        self._msg = [] # Put warning message here
+        self._trunc_limit = 1000 # To trunc submissions if there are too many submissions
+        self._time_alert_computation_stats = 2
+        
+        return course
+        
+    def POST(self, courseid):  # pylint: disable=arguments-differ
+        """ POST request """
+        course = self.init(courseid)
+
+        input = self.get_input()
+        tasks = course.get_tasks()
+        data, __ = self.get_submissions(course, input)
+
+        if "replay" in web.input():
+            if self.user_manager.has_admin_rights_on_course(course) == False:
+                raise web.notfound()
+            for submission in data:
+                self.submission_manager.replay_job(tasks[submission["taskid"]], submission)
+            self._msg.append(_("{0} selected submissions were set for replay.").format(str(len(data)))) 
+            
+        return self.page(course)
+
+    def GET_AUTH(self, courseid):  # pylint: disable=arguments-differ
+        """ GET request """
+        return self.page(self.init(courseid))
+
+    def submission_url_generator(self, submissionid):
+        """ Generates a submission url """
+        return "?submission=" + submissionid
+
+    def page(self, course):
+        """ Get all data and display the page """
+        
+        input = self.get_input()
+        data, classroom = self.get_submissions(course, input) #ONLY classrooms user wants to query
+        if len(data) == 0 and not self.show_collapse(input):
+            self._msg.append(_("No submissions found"))
+            
+        if "json" in input:
+            for submission in data:
+                submission = self.submission_manager.get_input_from_submission(submission)
+                #if need also feedback, uncomment this line
+                #submission = self.submission_manager.get_feedback_from_submission(submission,show_everything=True,translation=self.app._translations.get(self.user_manager.session_language(), gettext.NullTranslations()))
+            return data
+            
+        classrooms = self.user_manager.get_course_aggregations(course) # ALL classrooms of the course
+        users = self.get_users(course) # All users of the course
+        tasks = course.get_tasks();  # All tasks of the course
+
+        statistics = None
+        if input.stat != "no_stat":
+            old_time = time.time()
+            statistics = compute_statistics(tasks, data, True if "with_pond_stat" == input.stat else False)
+            total_time = time.time() - old_time
+            if total_time > self._time_alert_computation_stats:
+                self._msg.append(_("Statistics take {0} seconds to compute due to the higher number of submissions. Try adding more filter options to reduce computation time.").format(round(total_time,1)))
```

Comment:
Actually once this message is displayed the statistics are computed and displayed too...so it is quite useless.